CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/apis/OpenAiTests.cs

```diff
@@ -199,6 +201,76 @@ public enum Emotion { happy, sad, angry }
 
         }
 
+
+        [Fact]
+        public async Task ExampleTTSandSTT() {
+            string textToTest = "hello world";
+            Stream TTSResponseStream = await TTS(new Audio.TTSRequest() { input = textToTest });
+            Assert.NotNull(TTSResponseStream);
+
+            Audio.STTResponse STTResponse = await STT(new Audio.STTRequest() { file = TTSResponseStream });
+            Assert.NotEmpty(STTResponse.text);
+            Log.d(STTResponse.text);
+            //fails when textToTest contains numbers
+            Assert.Equal(formatString(STTResponse.text), formatString(textToTest));
+        }
+
+        private string formatString(string str) {
+            //helper function to format TTS and STT converted texts to test weather input text and output text is equal
+            return new String(str.ToCharArray().Where(c => !Char.IsWhiteSpace(c) && Char.IsLetterOrDigit(c))
+            .ToArray()).ToLower();
+        }
+
+        public async Task<Stream> TTS(Audio.TTSRequest requestParam) {
+            var openAiKey = await IoC.inject.GetAppSecrets().GetSecret("OpenAiKey");
+            return await new Uri("https://api.openai.com/v1/audio/speech").SendPOST().WithAuthorization(openAiKey)
+            .WithJsonContent(requestParam).GetResult<Stream>();
+        }
+
+        public async Task<Audio.STTResponse> STT(Audio.STTRequest requestParam) {
+            var openAiKey = await IoC.inject.GetAppSecrets().GetSecret("OpenAiKey");
+
+            Dictionary<string, object> formContent = new Dictionary<string, object>();
+            formContent.Add("model", requestParam.model);
+            formContent.Add("language", requestParam.language);
+            formContent.Add("responseFormat", requestParam.responseFormat);
+            formContent.Add("temperature", requestParam.temperature);
+            if (requestParam.prompt != null) {
+                formContent.Add("prompt", requestParam.prompt);
+            }
+
+            DirectoryEntry dir = EnvironmentV2.instance.GetNewInMemorySystem();
+            FileEntry fileToUpload = dir.GetChild("speech.mp3");
+            await fileToUpload.SaveStreamAsync(requestParam.file, resetStreamToStart: false);
```

Comment:
move these 3 lines here out of the STT method and then the sst method can also be moved into the OpenAi class (same as above)

Reply:
Same as above, all functionality of TTS and STT have been now moved to OpenAi class